src/analysis/special_functions/gamma.lean

```diff
@@ -1214,3 +1215,278 @@ end
+  rw [Gamma_seq, Gamma_seq, div_mul_div_comm, aux, ←pow_two],
+  have : (n : ℂ) ^ z * n ^ (1 - z) = n,
+  { rw [←cpow_add _ _ (nat.cast_ne_zero.mpr hn), add_sub_cancel'_right, cpow_one] },
+  rw [this, finset.prod_range_succ', finset.prod_range_succ, aux, ←finset.prod_mul_distrib,
+    nat.cast_zero, add_zero, add_comm (1 - z) n, ←add_sub_assoc],
+  have : ∀ (j : ℕ), (z + ↑(j + 1)) * (1 - z + ↑j) = ↑((j + 1) ^ 2) * (1 - z ^ 2 / (↑j + 1) ^ 2),
+  { intro j,
+    push_cast,
+    have : (j:ℂ) + 1 ≠ 0, by { rw [←nat.cast_succ, nat.cast_ne_zero], exact nat.succ_ne_zero j },
+    field_simp, ring },
+  simp_rw this,
+  rw [finset.prod_mul_distrib, ←nat.cast_prod, finset.prod_pow,
+    finset.prod_range_add_one_eq_factorial, nat.cast_pow,
+    (by {intros, ring} : ∀ (a b c d : ℂ), a * b * (c * d) = a * (d * (b * c))),
+    ←div_div, mul_div_cancel, ←div_div, mul_comm z _, mul_one_div],
+  exact pow_ne_zero 2 (nat.cast_ne_zero.mpr $ nat.factorial_ne_zero n),
+end
+
+lemma Gamma_neg_nat_eq_zero (n : ℕ) : Gamma (-n) = 0 :=
+begin
+  induction n with n IH,
+  { rw [Gamma, nat.cast_zero, neg_zero, zero_re, sub_zero, nat.floor_one, Gamma_aux],
+    simp only [div_zero] },
+  { have A : -(n.succ : ℂ) ≠ 0,
+    { rw [neg_ne_zero, nat.cast_ne_zero],
+      apply nat.succ_ne_zero },
+    have : -(n:ℂ) = -↑n.succ + 1, by simp,
+    rw [this, complex.Gamma_add_one _ A] at IH,
+    contrapose! IH,
+    exact mul_ne_zero A IH }
+end
+
+theorem Gamma_mul_Gamma_one_sub (z : ℂ) : Gamma z * Gamma (1 - z) = π / sin (π * z) :=
+begin
+  have pi_ne : (π : ℂ) ≠ 0, from complex.of_real_ne_zero.mpr pi_ne_zero,
+  by_cases hs : sin (↑π * z) = 0,
+  { -- first deal with silly case z = integer
+    rw [hs, div_zero],
+    rw [←neg_eq_zero, ←complex.sin_neg, ←mul_neg, complex.sin_eq_zero_iff, mul_comm] at hs,
+    obtain ⟨k, hk⟩ := hs,
+    rw [mul_eq_mul_right_iff, eq_false_intro (of_real_ne_zero.mpr pi_pos.ne'), or_false,
+      neg_eq_iff_neg_eq] at hk,
+    rw ←hk,
+    cases k,
+    { rw [int.cast_of_nat, complex.Gamma_neg_nat_eq_zero, zero_mul] },
+    { rw [int.cast_neg_succ_of_nat, neg_neg, nat.cast_add, nat.cast_one, add_comm, sub_add_cancel',
+        complex.Gamma_neg_nat_eq_zero, mul_zero] } },
+  refine tendsto_nhds_unique ((Gamma_seq_tendsto_Gamma z).mul (Gamma_seq_tendsto_Gamma $ 1 - z)) _,
+  have : ↑π / sin (↑π * z) = 1 * (π / sin (π * z)), by rw one_mul, rw this,
+  refine tendsto.congr' ((eventually_ne_at_top 0).mp
+    (eventually_of_forall (λ n hn, (Gamma_seq_mul z hn).symm))) (tendsto.mul _ _),
+  { convert tendsto_coe_nat_div_add_at_top (1 - z), ext1 n, rw add_sub_assoc },
+  { have : ↑π / sin (↑π * z) = 1 / (sin (π * z) / π), by field_simp, rw this,
+    refine tendsto_const_nhds.div _ (div_ne_zero hs pi_ne),
+    rw [←tendsto_mul_iff_of_ne_zero tendsto_const_nhds pi_ne, div_mul_cancel _ pi_ne],
+    convert tendsto_euler_sin_prod z,
+    ext1 n, rw [mul_comm, ←mul_assoc] },
+end
+
+theorem Gamma_ne_zero {s : ℂ} (hs : ∀ m:ℕ, s + m ≠ 0) : Gamma s ≠ 0 :=
```

Comment:
Possibly better as:
```suggestion
theorem Gamma_ne_zero {s : ℂ} (hs : ∀ m : ℕ, s ≠ -m) : Gamma s ≠ 0 :=
```

It would be great to state the `iff` version too; that `Gamma s = 0 \iff \exist m, s = -m`; at a glance it seems to be true for your definition